src/cljs/rems/user_settings.cljs

```diff
@@ -81,10 +81,30 @@
                    (when first-time? ; to avoid an infinite loop if the settings fail to save (e.g. unsupported language in cookies)
                      [[:check-if-should-save-language!]]))})))
 
-(defn fetch-user-settings! []
+(rf/reg-event-fx
+ :flash-message-error
+ (fn [_ user-settings-error]
+   ;; TODO: refactor this part
+   ;; this doesnt work
+   ;; (flash-message/default-error-handler :top "Fetch user settings")
+   ;; this works
+   (flash-message/show-default-error! :top (str "Fetch user settings" "wtf"))
+   {}))
+
+(rf/reg-event-fx
+ :loaded-user-settings-fail
+ (fn [{:keys [db]} [_ user-settings-error]]
+   (let [logged-in? (:user (:identity db))]
+     (if-not logged-in?
+       {:db db}
+       {:db db
+        :dispatch [:flash-message-error user-settings-error]}))))
+
+(defn fetch-user-settings! [opts]
```

Comment:
Good news, @opqdonut and I discussed this and Joel noticed that since we only use `fetch-user-settings!` for the initial fetch (and not for user settings page) we can skip these dispatch events and just ignore everything when the response code is 401. We can skip the logged-in part since this doesn't need really need the complication.

Reply:
I do not see the login part as a complication, but rather as a logical extension

Reply:
A logical extension that is not useful is basically unnecessary logic and code that should be simplified i.e. not exist?

Reply:
If you are talking about ```:flash-message-error```  then it has been addressed